crates/subspace-networking/src/node_runner.rs

```diff
@@ -272,8 +280,11 @@ where
             SwarmEvent::Behaviour(Event::PeerInfo(event)) => {
                 self.handle_peer_info_event(event).await;
             }
-            SwarmEvent::Behaviour(Event::ConnectedPeers(event)) => {
-                self.handle_connected_peers_event(event).await;
+            SwarmEvent::Behaviour(Event::GeneralConnectedPeers(event)) => {
+                self.handle_connected_peers_event(Left(event)).await;
+            }
+            SwarmEvent::Behaviour(Event::SpecialConnectedPeers(event)) => {
+                self.handle_connected_peers_event(Right(event)).await;
             }
```

Comment:
Please don't use `Left` and `Right` just to reuse the method, it looks somewhat ugly. Either inline the code or create two utility methods.